components/create_user_groups_modal/create_user_groups_modal.tsx

```diff
@@ -0,0 +1,265 @@
+        this.setState({usersToAdd});
+    };
+
+    private deleteUserCallback = (usersToAdd: UserProfile[]): void => {
+        this.setState({usersToAdd});
+    };
+
+    goBack = () => {
+        if (typeof this.props.backButtonCallback === 'function') {
+            this.props.backButtonCallback();
+            this.props.onExited();
+        }
+    }
+
+    createGroup = async (users?: UserProfile[]) => {
+        this.setState({showUnknownError: false, mentionInputErrorText: '', nameInputErrorText: ''});
+        let mention = this.state.mention;
+        const displayName = this.state.name;
+
+        if (displayName.length < 1) {
+            this.setState({nameInputErrorText: Utils.localizeMessage('user_groups_modal.nameIsEmpty', 'Name is a required field.')});
+            return;
+        }
+
+        if (!users || users.length === 0) {
+            return;
+        }
+        if (mention.substring(0, 1) === '@') {
+            mention = mention.substring(1, mention.length);
+        }
+
+        if (mention.length < 1) {
+            this.setState({mentionInputErrorText: Utils.localizeMessage('user_groups_modal.mentionIsEmpty', 'Mention is a required field.')});
+            return;
+        }
+
+        const mentionRegEx = new RegExp(/[^A-Za-z0-9]/g);
+        if (mentionRegEx.test(mention)) {
+            this.setState({mentionInputErrorText: Utils.localizeMessage('user_groups_modal.mentionInvalidError', 'Invalid character in mention.')});
+            return;
+        }
+
+        const group = {
+            name: mention,
+            display_name: this.state.name,
+            allow_reference: true,
+            source: 'custom',
+            user_ids: users.map((user) => {
+                return user.id;
+            }),
+        };
+
+        const data = await this.props.actions.createGroupWithUserIds(group);
+
+        if (data?.error) {
+            if (data.error?.server_error_id === 'app.custom_group.unique_name') {
+                this.setState({mentionInputErrorText: Utils.localizeMessage('user_groups_modal.mentionNotUnique', 'Mention needs to be unique.')});
+            } else {
+                this.setState({showUnknownError: true});
+            }
+        } else if (typeof this.props.backButtonCallback === 'function') {
+            this.goBack();
+        } else {
+            this.doHide();
+        }
+    }
+
+    render() {
+        return (
+            <Modal
+                dialogClassName='a11y__modal user-groups-modal-create'
+                show={this.state.show}
+                onHide={this.doHide}
+                onExited={this.props.onExited}
+                role='dialog'
+                aria-labelledby='createUserGroupsModalLabel'
+                id='createUserGroupsModal'
+            >
+                <Modal.Header closeButton={true}>
+                    {
+                        typeof this.props.backButtonCallback === 'function' ?
```

Comment:
Is there any reason to not use just `this.props.backButtonCallback ? ...`. Looks like the type definition assures that this is or a function or an `undefined`.

Reply:
Fixed

Reply:
Looks like it is not fixed